presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/WindowFilterPushDown.java

```diff
@@ -80,12 +80,18 @@ public PlanNode visitWindow(WindowNode node, RewriteContext<Constraint> context)
                 PlanNode rewrittenSource = context.rewrite(node.getSource(), null);
                 Optional<Integer> limit = getLimit(node, context.get());
                 if (node.getOrderBy().isEmpty()) {
-                    return new RowNumberNode(idAllocator.getNextId(),
+                    PlanNode rowNumberNode =  new RowNumberNode(idAllocator.getNextId(),
                             rewrittenSource,
                             node.getPartitionBy(),
                             getOnlyElement(node.getWindowFunctions().keySet()),
                             limit,
                             Optional.empty());
+                    if (limit.isPresent()) {
```

Comment:
Nice catch! I have fixed it by checking context to see if there is a `LIMIT` clause. See 2acdb3b20bb66e6201c0ed2ce8fc8833727eab8a